File: app/src/main/java/com/xephorium/crystalnote/ui/util/KeyboardUtils.java
package com.xephorium.crystalnote.ui.util;

import android.app.Activity;
import android.content.Context;
import android.view.View;
import android.view.inputmethod.InputMethodManager;

public final class KeyboardUtils {

    public static void hideKeyboard(Activity activity) {
        View view = activity.getCurrentFocus();
        if (view != null) {
            InputMethodManager imm = (InputMethodManager) activity.getSystemService(Context.INPUT_METHOD_SERVICE);
            imm.hideSoftInputFromWindow(view.getWindowToken(), 0);
        }
    }

    public static void showKeyboard(Activity activity) {
        View view = activity.getCurrentFocus();
        if (view != null) {
            InputMethodManager imm = (InputMethodManager) activity.getSystemService(Context.INPUT_METHOD_SERVICE);
            imm.showSoftInput(view, 0);
        }
    }

    public static int getKeyboardHeight(Activity activity) {
        int appHeight = DisplayUtils.getAppHeight(activity);
        int displayHeight = DisplayUtils.getDisplayHeight(activity);
        int statusBarHeight = DisplayUtils.getStatusBarHeight(activity);
        return displayHeight - (appHeight + statusBarHeight);
    }
}

File: app/src/main/java/com/xephorium/crystalnote/ui/utility/KeyboardUtils.kt
package com.xephorium.crystalnote.ui.utility

import android.app.Activity
import android.content.Context
import android.view.inputmethod.InputMethodManager

object KeyboardUtils {


    /*--- Public Utility Methods ---*/

    fun hideKeyboard(activity: Activity) {
        val view = activity.currentFocus
        if (view != null) {
            val imm = activity.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            imm.hideSoftInputFromWindow(view.windowToken, 0)
        }
    }

    fun showKeyboard(activity: Activity) {
        val view = activity.currentFocus
        if (view != null) {
            val imm = activity.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
            imm.showSoftInput(view, 0)
        }
    }

    fun getKeyboardHeight(activity: Activity): Int {
        val appHeight = DisplayUtils.getAppHeight(activity)
        val displayHeight = DisplayUtils.getDisplayHeight(activity)
        val statusBarHeight = DisplayUtils.getStatusBarHeight(activity)
        return displayHeight - (appHeight + statusBarHeight)
    }
}
